code/game/gamemodes/abduction/abduction_gear.dm

```diff
@@ -0,0 +1,251 @@
+//AGENT VEST
+/obj/item/clothing/suit/armor/abductor/vest
+	name = "Agent Vest"
+	desc = "Alien vest."
+	icon_state = "armor"
+	item_state = "armor"
+	blood_overlay_type = "armor"
+	armor = list(melee = 50, bullet = 15, laser = 50, energy = 10, bomb = 25, bio = 0, rad = 0)
+	action_button_name = "Activate"
+	action_button_is_hands_free = 1
+	var/mode = "STEALTH"
+	var/stealth_active = 0
+	var/combat_cooldown = 10
+	var/datum/icon_snapshot/disguise
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/flip_mode()
+	switch(mode)
+		if("STEALTH")
+			mode = "COMBAT"
+			DeactivateStealth()
+			armor = list(melee = 50, bullet = 50, laser = 50, energy = 50, bomb = 50, bio = 50, rad = 50)
+			//TODO Update looks
+			return
+		if("COMBAT")// TO STEALTH
+			mode = "STEALTH"
+			armor = list(melee = 15, bullet = 15, laser = 15, energy = 15, bomb = 15, bio = 15, rad = 15)
+			//TODO Update looks
+			return
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/SetDisguise(var/datum/icon_snapshot/entry)
+	disguise = entry
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/ActivateStealth()
+	if(disguise == null)
+		return
+	stealth_active = 1
+	if(istype(src.loc, /mob/living/carbon/human))
+		var/mob/living/carbon/human/M = src.loc
+		spawn(0)
+			anim(M.loc,M,'icons/mob/mob.dmi',,"cloak",,M.dir)
+
+		M.name = disguise.name
+		M.icon = disguise.icon
+		M.icon_state = disguise.icon_state
+		M.overlays = disguise.overlays //TODO Ignore Inhands ?
+		M.update_inv_r_hand()
+		M.update_inv_l_hand()
+	return
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/DeactivateStealth()
+	if(!stealth_active)
+		return
+	stealth_active = 0
+	if(istype(src.loc, /mob/living/carbon/human))
+		var/mob/living/carbon/human/M = src.loc
+		spawn(0)
+			anim(M.loc,M,'icons/mob/mob.dmi',,"uncloak",,M.dir)
+		M.name = M.real_name
+		M.overlays.Cut()
+		M.regenerate_icons()
+	return
+
+/obj/item/clothing/suit/armor/abductor/vest/IsShield()
+	DeactivateStealth()
+	return 0
+/obj/item/clothing/suit/armor/abductor/vest/IsReflect()
+	DeactivateStealth()
+	return 0
+
+/obj/item/clothing/suit/armor/abductor/vest/ui_action_click()
+	switch(mode)
+		if("COMBAT")
+			Adrenaline()
+		if("STEALTH")
+			if(stealth_active)
+				DeactivateStealth()
+			else
+				ActivateStealth()
+
+/obj/item/clothing/suit/armor/abductor/vest/proc/Adrenaline()
+	if(istype(src.loc, /mob/living/carbon/human))
+		if(combat_cooldown != initial(combat_cooldown))
+			src.loc << "<span class='warning'>Combat injection recharging</span>"
+		var/mob/living/carbon/human/M = src.loc
+		M.stat = 0
+		M.SetParalysis(0)
+		M.SetStunned(0)
+		M.SetWeakened(0)
+		M.lying = 0
+		M.update_canmove()
+		M.adjustStaminaLoss(-75)
+		combat_cooldown = 0
+		SSobj.processing |= src
+
+/obj/item/clothing/suit/armor/abductor/vest/process()
+	combat_cooldown++
+	if(combat_cooldown==initial(combat_cooldown))
+		SSobj.processing.Remove(src)
+
+/datum/icon_snapshot
```

Comment:
I'd like it if this could have a ```/datum/icon_snapshot/proc/makeImage()``` proc which takes the data in it (name, icon, icon_state, overlays) and produces a /image object and returns it. This is optional but this helper datum has uses outside of this.

(Probably also best to move it to the helper datum folder)